ginivision/src/main/java/net/gini/android/vision/internal/pdf/RendererLollipop.java

```diff
@@ -78,10 +85,23 @@ private PdfRenderer getPdfRenderer() {
             return new PdfRenderer(fileDescriptor);
         } catch (final IOException e) {
             LOG.error("Could not read pdf", e);
+            closeFileDescriptor(fileDescriptor);
+        } catch (final SecurityException e) {
+            LOG.error("Could not read pdf", e);
+            closeFileDescriptor(fileDescriptor);
+            throw e;
```

Comment:
Why throwing the exception again if it is handled here? The only thing that is done [with it](https://github.com/gini/gini-vision-lib-android/pull/222/files#diff-50519542387040bda5205076d7e2d4d7R55) is to print the same error in the log.

Reply:
To be able to catch the security exception in `isPdfPasswordProtected()`.